src/test/scala/com/github/mrpowers/spark/fast/tests/DatasetComparerTest.scala

```diff
@@ -20,7 +21,26 @@ class DatasetComparerTest extends AnyFreeSpec with DatasetComparer with SparkSes
     import spark.implicits._
 
     "provides a good README example" in {
+      val sourceDS = Seq(
+        Person("juan", 5),
+        Person("bob", 1),
+        Person("li", 49),
+        Person("alice", 5)
+      ).toDS
+
+      val expectedDS = Seq(
+        Person("juan", 5),
+        Person("frank", 10),
+        Person("li", 49),
+        Person("lucy", 5)
+      ).toDS
+
+      val e = intercept[DatasetContentMismatch] {
+        assertSmallDatasetEquality(sourceDS, expectedDS)
+      }
+    }
 
+    "Correctly mark unequal elements" in {
```

Comment:
@SemyonSinchenko , @alfonsorr - I added two small tests to validate output message of assertion. We can use ```extractColorGroup``` to add more robust testing on the message. I will create separate issue for improving tests.